src/makeInfinityAdder.js

```diff
@@ -4,7 +4,25 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let totalSum = 0;
+
+  function add(num) {
+    if (num !== undefined) {
+      totalSum += num;
+
+      return add;
+    }
+
+    if (num === undefined) {
```

Comment:
The condition `num === undefined` is unnecessary because the previous `if` statement already checks for the opposite condition. You can remove this `if` statement.